.murdock

```diff
@@ -242,8 +248,36 @@ get_apps() {
 # Only print for boards in $BOARDS.
 get_supported_boards() {
     local appdir=$1
+    local only_changed=0
+
+    if [ -n "$APPS_CHANGED" ]; then
+        if is_in_list "$appdir" "${APPS_CHANGED}"; then
+            # this app has changed -> build for all boards
+            true
+        else
+            # this is not a changed app -> build for changed boards
+            only_changed=1
+        fi
+    else
+        if [ -n "$BOARDS_CHANGED" ]; then
+            echo d
+            # no changed apps, some changed boards -> build for changed boards
+            only_changed=1
+        else
+            # no changed apps list, no changed boards list -> build all boards
+            true
+        fi
+    fi
+
+    if [ $only_changed -eq 1 ]; then
+        BOARDS_=${BOARDS}
+        export BOARDS="$BOARDS_CHANGED"
```

Comment:
Shouldn't the intersection of the supported boards and the changed boards be created here? Or is this done elsewhere? I fear that otherwise an app might be build for a board that is not supported.

Reply:
This exports "BOARDS" to the changed boards, which "make info-boards-supported" uses to generate the actual list.